packages/http-client/src/http-request-config.spec.ts

```diff
@@ -24,3 +33,73 @@ describe('filterCustomRequestConfig', () => {
     );
   });
 });
+
+describe('buildHttpRequestConfig', () => {
+  it('should pick headers and params with higher priorities', () => {
+    const withOrigin: HttpRequestConfigWithOrigin = {
+      method: 'get',
+      headers: {
+        DestinationProperty: { Authorization: 'destProp' },
+        RequestConfig: { Authorization: 'reqConfig' }
+      },
+      params: {
+        Custom: { param: 'custom' },
+        RequestConfig: { param: 'reqConfig' }
+      }
+    };
+
+    const expected: HttpRequestConfig = {
+      method: 'get',
+      headers: { Authorization: 'destProp' },
+      params: { param: 'custom' }
+    };
+    expect(buildHttpRequestConfig(withOrigin)).toStrictEqual(expected);
+  });
+});
+
+describe('getOptionWithPriority', () => {
+  it('should merge options', () => {
+    const originOptions: OriginOptions = {
+      Custom: { Authorization: 'customAuth' },
+      Destination: { 'sap-client': '001' },
+      RequestConfig: { 'if-match': 'etag' }
+    };
+    const expected = {
+      Authorization: 'customAuth',
+      'sap-client': '001',
+      'if-match': 'etag'
+    };
+    expect(mergeOptionsWithPriority(originOptions)).toStrictEqual(expected);
+  });
+
+  it('should use options with higher priority', () => {
+    const originOptions: OriginOptions = {
+      Custom: { param1: 'customParam1' },
+      DestinationProperty: {
+        param1: 'destPropParam1',
+        param2: 'destPropParam2'
+      },
+      Destination: { param2: 'destParam2', param3: 'destParam3' },
+      RequestConfig: { param3: 'reqParam3', param4: 'reqParam4' }
+    };
+    const expected = {
+      param1: 'customParam1',
+      param2: 'destPropParam2',
+      param3: 'destParam3',
+      param4: 'reqParam4'
+    };
+    expect(mergeOptionsWithPriority(originOptions)).toStrictEqual(expected);
+  });
+
+  it('should use options with higher priority and ignore case', () => {
+    const originOptions: OriginOptions = {
+      Custom: { Authorization: 'customAuth' },
+      Destination: { authorization: 'destAuth', 'sap-client': '001' }
```

Comment:
Minor: perhaps make hte casing difference stronger so that one sees it quicker: authorization versus AuThORizAtioN  or so.